composer.lock

```diff
@@ -1,54 +1,364 @@
 {
```

Comment:
Is this file necessary? Once we update the dependencies, this file will be updated again.

I think it's fine to remove this locked file from `Git` version control and avoid caching the dependencies.

Reply:
The lock file keeps the dependencies locked to the exact version they were at the time everything passed, so it's an extra precaution to ensure things don't break. This is an article that explains it better than I can: https://www.engineyard.com/blog/composer-its-all-about-the-lock-file

Here's the relevant section:

> Do You Know Where Your Lock File Is?
> A few weeks ago on Twitter, I noticed that the OpenCFP project doesn’t have a composer.lock file in it’s repository. “So what,” you might say, “just composer install and away you go. You’ll get the same dependencies, right?”
> 
> Wrong.
> 
> The point of the lock file is to record the exact versions that are installed so they can be re-installed. This means that if you have a version spec of 1.* and your co-worker runs composer update which installs 1.2.4, and then commits the composer.lock file, when you composer install, you will also get 1.2.4, even if 1.3.0 has been released. This ensures everybody working on the project has the same exact version.
> 
> Now, you might be thinking: with proper semantic versioning, 1.3.0 (or even 1.2.5) should be backwards compatible because it still conforms to the 1.* version spec, i.e. the major version number hasn’t changed. So what’s the big deal?
> 
> Well, unless you are performing code reviews on every release of every dependency, and their dependencies, and so on and so forth, it’s highly likely that some sort of bug or change will creep in eventually that will break your code. We’re only human, after all.
> 
> It’s also pretty common for a project to set a dependency version to rely on dev-master, meaning they pull the most recent changes each time. This means that if anything has been committed since the last time a composer install was done, then, without a lock file, you will get new third-party code being pulled down.
> 
> Again, this is a problem if you’re concerned about your code breaking. And it’s one of the reasons why it’s important to think about Composer as being centered around the composer.lock file. It is a safety device, and should be used like one.